consoles/virtio_terminal.pm

```diff
@@ -87,23 +87,37 @@ sub disable {
 sub save_snapshot {
     my ($self, $name) = @_;
 
-    if (defined($self->{screen})) {
-        $self->{snapshots}->{$name} = $self->{screen}->peak();
-    } else {
-        $self->{snapshots}->{$name} = '';
-    }
+    $self->set_snapshot($name, 'activated', $self->{activated});
+    $self->set_snapshot($name, 'buffer',    $self->{screen} ? $self->{screen}->peak() : $self->{preload_buffer});
 }
 
 sub load_snapshot {
     my ($self, $name) = @_;
 
+    $self->{activated} = $self->get_snapshot($name, 'activated') // 0;
+    my $buffer = $self->get_snapshot($name, 'buffer') // '';
     if (defined($self->{screen})) {
-        $self->{screen}->{carry_buffer} = $self->{snapshots}->{$name};
+        $self->{screen}->{carry_buffer} = $buffer;
     } else {
-        $self->{preload_buffer} = $self->{snapshots}->{$name};
+        $self->{preload_buffer} = $buffer;
     }
 }
 
+sub get_snapshot {
```

Comment:
When `$key` is `undef`, `get_snapshot($name)` returns the entire snapshot, but `set_snapshot($name)` is a noop. Shouldn't be an issue in practice, but might be a trap.

Reply:
I thought it might be good to get the full snapshot for what ever reason...

Anyway, if we think these setter/getters are over engineering I can also remove them. The reason why I kept it was:
* if we will implement snapshot in other consoles, it could help to move it back to `consoles::console`
* I didn't need to touch the tests again (no strong reason, I know :)

Reply:
IMO the getters and setters are fine, but IMO it should just assume the key is always defined.